### PR TITLE
Feature/tempfile

### PR DESCRIPTION
**Pull request by antoyo**
Based of pr #2.

### CODE REVIEW

File: src/fs.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+impl TempFile {
+    pub fn new() -> io::Result<Self> {
+        let mut rng = Rng::new();
+        let mut path = None;
+        for _ in 0..50 {
+            let tempfile = temp_dir()
+                .join(format!("file{}", rng.next()));
+            if !tempfile.exists() {
```
Comment (gnuvince) [inline src/fs.rs]: It'd be cool if we could pass the prefix so that a temp file for Prophet is called `prophet.1234` and one for ACR is called `acr.6789`.

Comment (antoyo) [reply src/fs.rs]: Done.

File: src/fs.rs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    pub fn with_prefix(prefix: &str) -> io::Result<Self> {
+        let mut rng = Rng::new();
+        let mut path = None;
+        for _ in 0..50 {
+            let tempfile = temp_dir()
+                .join(format!("{}.{}", prefix, rng.next()));
+            if let Ok(_) = OpenOptions::new()
+                .write(true)
+                .create_new(true)
+                .open(&tempfile)
```
Comment (gnuvince) [inline src/fs.rs]: Can you specify permissions?  0600 (read+write for owner, nothing for group and others) would be ideal.

Comment (gnuvince) [reply src/fs.rs]: https://doc.rust-lang.org/stable/std/os/unix/fs/trait.PermissionsExt.html#tymethod.set_mode

**Pull request merged**